docs/sources/layers/recurrent.md

```diff
@@ -128,6 +128,39 @@ Long-Short Term Memory unit - Hochreiter 1997.
     - [Long short-term memory](http://deeplearning.cs.cmu.edu/pdfs/Hochreiter97_lstm.pdf) (original 1997 paper)
     - [Learning to forget: Continual prediction with LSTM](http://www.mitpressjournals.org/doi/pdf/10.1162/089976600300015015)
     - [Supervised sequence labelling with recurrent neural networks](http://www.cs.toronto.edu/~graves/preprint.pdf)
+
+---
+
+## MutatedRNN_1, MutatedRNN_2, MutatedRNN_3
+
+```python
+keras.layers.recurrent.MutatedRNN_1(input_dim, output_dim=128, 
+        init='glorot_uniform', inner_init='orthogonal', 
+        activation='tanh', inner_activation='sigmoid',
+        weights=None, truncate_gradient=-1, return_sequences=False)
+```
+
+Top 3 RNN architectures evolved from the evaluation of thousands of models. Serves as alternatives to LSTMs and GRUs. Corresponds to `MUT1`, `MUT2`, and `MUT3` architectures described in the paper: An Empirical Exploration of Recurrent Network Architectures, Jozefowicz et al. 2015. Note that `input_dim` and `output_dim` must be equal in the `MUT1` and `MUT2` architectures.
```

Comment:
In the equations some weight matrices are dropped. For example reset in `MUT2` is governed by `x_t + W_hr * h_tm1 + b_r`, as compared to `W_xr * x_t + W_hr * h_tm1 + b_r` in GRU. I suppose changing dimensionality would require another layer. This constraint may not be entirely in the spirit of keras, but they do perform quite well so I think it's worth including.

Right now an exception is raised if `input_dim` and `output_dim` are not equal, in the case of the first 2 architectures. Do you think it's better to have a single `dim` parameter?